create-tiny-url.go

```
@@ -0,0 +1,90 @@
+package main
+
+import (
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+type UrlObject struct {
+	Url string
+}
+
+type UrlStoredObject struct {
+	Url  string
+	Tiny string
+}
+
+type UrlStoredObjectSet []UrlStoredObject
+
+var filename string = "./url_mappings.json"
+
+func generateTinyUrl(w http.ResponseWriter, r *http.Request) {
+	var urlObject UrlObject
+	reqBody, _ := ioutil.ReadAll(r.Body)
```

1)  always check for erorrs. It's a bad practice to hide errors.